libarchive/archive_write_set_format_xar.c

```diff
@@ -3417,6 +3422,267 @@ xml_writer_get_final_content_and_length(struct xml_writer *ctx,
 	*size = (size_t)ctx->bp->use;
 	return (0);
 }
+
+#elif HAVE_XMLLITE_H
+
+struct xml_writer {
```

Comment:
Did you forget to `git add` the part which calls into this code - as-is it seems unused. The reader part seemed fine.

Cannot see any calls to `archive_set_error()` in the writer, is that intentional?

Reply:
FWIW, the writer was wired up by e619342dfa36b887ffa0ea33e98d04cb161cd7de introducing `#ifdef`-replaceable implementations of `xml_writer_create` et al. The tests running in the MSVC and mingw builds passing confirms this (rather than them being marked skipped 🙂)

Reply:
As for `archive_set_error`, good catch. I am not sure how I missed that. :)